src/main/java/TaskList.java

```diff
@@ -0,0 +1,124 @@
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class TaskList {
+
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+    public enum TaskType {
+        DEADLINE,
+        EVENT,
+        TODO
+    }
+
+    public void addToList(TaskType taskType, String description, String deadline)
+            throws IllegalArgumentException {
+        Task newTask = null; // task to be added
+
+        // based on task type, create corresponding subclass object
+        switch (taskType) {
+        case DEADLINE:
+            newTask = new Deadline(description, deadline);
+            break;
+        case EVENT:
+            newTask = new Event(description, deadline);
+            break;
+        case TODO:
+            newTask = new Todo(description);
+            break;
+        default:
+            // this should not happen; invalid task type should be caught in caller method already
+            System.out.println("Something went wrong!");
+            break;
+        }
+
+        // add newTask to taskList
+        tasks.add(newTask);
+        System.out.println("Okay! I've added: " + description
+                + ". Use list to see all your tasks!");
+        // todo: description has trailing space due to using /at or /by as delimiter. how?
```

Comment:
Use `strip()`, `stripLeading()`, `stripTrailing()`, whichever that suits your need :)